src/main/java/org/elasticsearch/common/io/FileSystemUtils.java

```diff
@@ -205,14 +208,42 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
                     // We just move the new file to new dir
                     Files.move(file, path);
```

Comment:
To be coherent with the `preVisitDirectory()` method I think we should use the `FileSystemUtils.move()` method instead of Files.move() that doesn't support moving non-empty directories between different filesystems.